packages/graphql-hooks/src/middlewares/README.md

```diff
@@ -0,0 +1,58 @@
+# GraphQLClient Middleware
+
+A middleware gives you the option to access and intercept requests before and after they are performed.
+
+In GraphQLClient the middleware functions are passed in the GraphQLClient's config under the `middleware` key. They are executed in the given order.
+
+## Minimal example
+
+```js
+const LoggerMiddleware = ({ operation }, next) => {
+    console.log('Starting request:', operation)
+    next()
+  }
+}
+```
+
+## Parameters
+
+Middleware is a function with 2 parameters:
+
+- `options` object:
+  - `operation` - GraphQL operation object - includes `query`, `variables` etc.
+  - `addResponseHook` - a function that can be called from within the middleware to add a response hook - it accepts a handler to read/transform and return the data - passthrough ex.: `addResponseHook(response => response)`
+  - `client` - the GraphQLClient instance
+  - `resolve`, `reject` - advanced usage only, these would be used only when we want to resolve/reject the request `Promise` early without doing the usual fetch (see [example](examples/cacheMiddleware.js))
+- `next` function - calls the next middleware function in line. Generally it should always be called, unless we want to change the control flow
+
+## More examples
+
+See [examples folder](examples/)
+
+### Update response post-request
+
+```js
+const camelCaseMiddleware = ({ operation, addResponseHook }, next) => {
+  addResponseHook(response => {
+    // Need to return
+    return toCamelCaseDeep(response)
```

Comment:
what would this do? doesn't feel like a typical operation somebody would want to do. maybe a more realistic example?

Reply:
Right, I had a use case for something similar with rest api that returned snake case vars and later were camel cased so got inspiration in there. Will think of something more realistic 👍

Reply:
thanks. not a big deal, let's just make sure it's clear what we're trying to do